theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-toolbar-contribution.tsx

```diff
@@ -114,28 +146,57 @@ export class TraceViewerToolbarContribution extends AbstractViewContribution<Wid
                     });
                 }}></div>
             </div>,
-            priority: 1,
+            priority: 4,
             group: 'navigation',
             onDidChange: this.onAnnotationsChangedEvent,
         });
-
-        registry.registerItem({
-            id: TraceViewerToolbarCommands.ZOOM_IN.id,
-            command: TraceViewerToolbarCommands.ZOOM_IN.id,
-            tooltip: TraceViewerToolbarCommands.ZOOM_IN.label,
-            priority: 1,
-        });
-        registry.registerItem({
-            id: TraceViewerToolbarCommands.ZOOM_OUT.id,
-            command: TraceViewerToolbarCommands.ZOOM_OUT.id,
-            tooltip: TraceViewerToolbarCommands.ZOOM_OUT.label,
-            priority: 2,
-        });
         registry.registerItem({
-            id: TraceViewerToolbarCommands.RESET.id,
-            command: TraceViewerToolbarCommands.RESET.id,
-            tooltip: TraceViewerToolbarCommands.RESET.label,
-            priority: 3,
+            id: TraceViewerToolbarCommands.MARKERS.id,
+            isVisible: w => w instanceof TraceViewerWidget && this.markersSetMap.size > 0,
+            // render() here is not a react component and hence need to disable the react display-name rule
+            // eslint-disable-next-line react/display-name
+            render: () => <div className="item enabled">
+                <div id="trace.viewer.toolbar.filter" className="fa fa-bars" title="Markers filter" onClick={async (event: React.MouseEvent) => {
+                    const toDisposeOnHide = new DisposableCollection();
+                    const menuPath = TraceViewerToolbarMenus.MARKERS_MENU;
+                    let index = 0;
+                    this.markersSetMap.forEach((toggleInd, marker, self) => {
+                        index += 1;
+                        toDisposeOnHide.push(this.menus.registerMenuAction(menuPath, {
+                            label: marker.name,
+                            commandId: marker.name.toString() + index.toString(),
+                            order: index.toString(),
+                        }));
+
+                        toDisposeOnHide.push(this.commands.registerCommand({
+                            id: marker.name.toString() + index.toString(),
+                            label: marker.name
+                        }, {
+                            execute: () => {
+                                const selectInd = self.get(marker);
+                                const selectedMarkerSet = Array.from(self.keys()).find(markerSet => self.get(markerSet) === true);
+                                if (selectedMarkerSet) {
+                                    self.set(selectedMarkerSet, false);
+                                }
+                                self.set(marker, !selectInd);
+                                signalManager().fireMarkerSetSelectedSignal(selectInd ? '' : marker.id);
```

Comment:
Not sure where it should happen, but when you select a new marker set, it should trigger a new fetchAnnotationCategories so that the Markers filter menu is updated with the new categories.

Reply:
The api to retrieve annotation categories doesn't accept markerSetId. 

https://theia-ide.github.io/trace-server-protocol/#operation/getMarkerSets

Right now, the markerSetId is passed only to fetch annotations and not categories. Is this incorrect?